toolbox/timecomparer/main.go

```diff
@@ -0,0 +1,532 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	con "context"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"reflect"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/golang-collections/collections/stack"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/net/context"
+	"google.golang.org/api/option"
+	"google.golang.org/api/sheets/v4"
+)
+
+type TimeComparer struct {
+	client     *storage.Client
+	bucketName string
+}
+
+// Read google spreadsheets with credentials, spreadsheet ID and read range to get slice of file paths to build-logs.
+func readSpreadSheet(ctx con.Context, apiKey string, spreadsheetID string, readRange string) []string {
+	srv, err := sheets.NewService(ctx, option.WithAPIKey(apiKey))
+	if err != nil {
+		log.Fatalf("Unable to retrieve Sheets client: %v", err)
+	}
+
+	resp, err := srv.Spreadsheets.Values.Get(spreadsheetID, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v", err)
+	}
+
+	if len(resp.Values) == 0 {
+		fmt.Println("No data found.")
+		return nil
+	}
+	filePaths := []string{}
+
+	for _, row := range resp.Values {
+		filePath := row[0]
+		filePathString, err := filePath.(string)
+		if err {
+			filePaths = append(filePaths, filePathString)
+		}
+	}
+
+	return filePaths
+}
+
+func NewTimeComparer(client *storage.Client, bucketName string) *TimeComparer {
+	return &TimeComparer{
+		client:     client,
+		bucketName: bucketName,
+	}
+}
+
+// Read from gcs the build-log.txt files into slice of lines.
+func (tc *TimeComparer) query(ctx context.Context, prefix string) ([]string, error) {
+	client := tc.client
+	bucket := client.Bucket(tc.bucketName)
+	buildFile := bucket.Object(prefix + "build-log.txt")
+
+	rc, err := buildFile.NewReader(ctx)
+	if err != nil {
+		return []string{}, err
+	}
+	defer rc.Close()
+	lines := []string{}
+
+	scanner := bufio.NewScanner(rc)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) != 0 {
```

Comment:
In general try to minimize indentation and return early.

So 
```golang
if len(line) == 0 {
  continue
}
lines = append(lines, line)
```

has lower indentation than the current